debugger/src/main/kotlin/org/rust/debugger/prettyPrinters/lldb_pretty_printer.py

```diff
@@ -0,0 +1,67 @@
+"""
+You can find some useful information here:
+    1. https://lldb.llvm.org/varformats.html
+    2. https://lldb.llvm.org/python-reference.html
+"""
+
+
+def size_as_summary(valobj):
+    return 'size=' + str(valobj.GetNumChildren())
+
+
+def SizeSummaryProvider(valobj, dict):
+    return size_as_summary(valobj)
+
+
+class StdVecProvider:
+    """
+    Synthetic children is the way to provide a children-based user-friendly representation of the object's value.
+    To create synthetic children, you need to provide a class that "implements" a given interface:
+
+        class SyntheticChildrenProvider:
+            def __init__(self, valobj: SBValue, internal_dict)
+            def num_children(self)
+            def get_child_index(self, name: String)
+            def get_child_at_index(self, index: Int)
+            def update(self)
+            def has_children(self)
+            def get_value(self)
+
+    `internal_dict` is an internal support parameter used by LLDB and you should not touch it;
+    `valobj` is the object encapsulating the actual variable being displayed;
+    """
+
+    def __init__(self, valobj, internal_dict):
+        self.valobj = valobj
+        self.length = valobj.GetChildMemberWithName("len").GetValueAsUnsigned()
+        self.buf = valobj.GetChildMemberWithName("buf")
+        self.data_ptr = self.buf.GetChildAtIndex(0).GetChildAtIndex(0).GetChildAtIndex(0)
+        self.capacity = self.buf.GetChildAtIndex(1).GetValueAsUnsigned()
+        self.element_type = self.data_ptr.GetType().GetPointeeType()
+        self.element_type_size = self.element_type.GetByteSize()
+
+    def num_children(self):
+        return self.length
+
+    def get_child_index(self, name):
+        try:
+            return int(name.lstrip('[').rstrip(']'))
+        except:
```

Comment:
catchall usually is considered a bad style. I thin it’s better to use name.isdigit check?